torch_xla/distributed/xla_dist.py

```diff
@@ -294,6 +335,13 @@ def _cleanup_worker(local_script, remote_script, client_worker):
           ))
       thread.start()
       threads.append(thread)
+
+    # Cleanup states in case of restart
```

Comment:
With this pattern, you have the same code in the constructor, and here.
Have an init() (or something) function that you call in two places.